daemons/fenced/fenced_remote.c

```diff
@@ -1597,7 +1597,11 @@ get_op_total_timeout(const remote_fencing_op_t *op,
         total_timeout = op->base_timeout;
     }
 
-    return total_timeout ? total_timeout : op->base_timeout;
+    /* Take any requested fencing delay into account to prevent it from eating
+     * up the total timeout.
+     */
+    return ((total_timeout ? total_timeout : op->base_timeout)
+            + (op->delay > 0 ? op->delay : 0));
```

Comment:
See https://projects.clusterlabs.org/T93 and https://bugs.clusterlabs.org/show_bug.cgi?id=5493

When I initially looked into this issue, my conclusion was that the fencer wouldn't calculate and report the total timeout until after the delay had already expired, so the delay should be taken into account on the controller's side (maybe in addition to here?). I may have been wrong, though.

Reply:
OTOH, it's not necessary but also not wrong that the initial timeout for fencing callback takes any requested fencing delay into account.